utils/src/main/java/org/corfudb/utils/lock/LockClient.java

```diff
@@ -160,7 +161,12 @@ private void monitorLocks() {
                             log.debug("LockClient: lease revoked for lock {}", lockId.getLockName());
                             locks.get(lockId).input(LockEvent.LEASE_REVOKED);
                         }
-                    } catch (Exception ex) {
+                    }
+                    catch (WrongClusterException wce) {
+                        log.error("Lock client {} is connected to a wrong cluster:", clientId, wce);
+                        throw wce;
+                    }
+                    catch (Exception ex) {
```

Comment:
we should log something here also

Reply:
Actually, if we re-throw an exception we must not log it.
In the other case, we will log the same thrown exceptions on all levels of the application, which dramatically increases log size and make it difficult to read and understand the application logic.

Would be great if Pavel removes those logs in `catch` block because they will be handled on a higher level of the stack